src/KafkaFlow/Extensions/ConfigurationBuilderExtensions.cs

```diff
@@ -82,6 +82,22 @@ public static IConsumerConfigurationBuilder WithConsumerConfig(this IConsumerCon
             return builder;
         }
 
+        /// <summary>
+        /// Adds a handler to be executed when KafkaFlow has pending offsets to be committed
+        /// </summary>
+        /// <param name="builder">The configuration builder</param>
+        /// <param name="pendingOffsetsHandler">A handler for the consumer pending offsets state</param>
+        /// <param name="intervalMs">The execution interval in milliseconds</param>
+        /// <returns></returns>
+        public static IConsumerConfigurationBuilder OnPendingOffsets(
```

Comment:
Why an extension method instead of a common method?

Reply:
it couldn't be in the abstraction layer because of Confluent.Kafka.TopicPartitionOffset Action parameter